### ISSUE TITLE
Is waiting on a already completed future a wait failure?

### ISSUE THREAD

**Issue by rdhananjaya** [ballerina-platform/ballerina-spec#802]
**Description:**
I'm trying to understand how a wait can fail.

Is waiting on an already completed future a wait failure? If so, how can a user differenciate an error due to a wait failure vs future returning a error value.

**Suggested Labels:**

   
**Code sample that shows issue:**

**Related Issues:**

**Comment by jclark** [comment]
> Is waiting on an already completed future a wait failure?

Yes: that's what the bit you quoted from 7.3.2 says "Only one wait on a strand can succeed; this wait receives the value returned by the strand. Any other waits on that strand fail."

> If so, how can a user differenciate an error due to a wait failure vs future returning a error value.

See #689.

**Comment by riyafa** [comment]
For this in the case of alternate wait action should the following return an error for  the result of `r1` because we try to wait on f4 twice?
```
    future<int|error> f1 = start calculate(3, 4);
    future<int|error> f4 = start calculate(9, 8);
    future<int|error> f5 = start calculate(5, 4);
    int|error result = wait f4|f5;
    int|error r1 = wait f4|f1;
    if(r1 is error) {
        panic r1;
    }
```

**Comment by jclark** [comment]
@riyafa Let's make this a separate issue
